To be merged 2.8.0 - MTV-2033 RN 2.8.0

Pkylas007 opened the pull request:
### JIRA

* [MTV-2033 Release Notes 2.8.0](https://issues.redhat.com/browse/MTV-2033)

### VERSION

* MTV 2.8.0

### PREVIEW

* [MTV 2.8.0 Release Notes](https://file.corp.redhat.com/pkylasam/mtv-2033-rn/html-single/)

### QE review:
- [ ] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Pkylas007 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,23 @@
+.Disk counter does not update progress while migrating virtual machine with multiple disks
+In earlier releases of {project-short}, the `Disk Counter`did not show real-time progress of disk transfer when migrating a virtual machine with multiple disks. This issue has been resolved in {project-short} 2.8.0.link:https://issues.redhat.com/browse/MTV-1718[(MTV-1718)]
+
+.Incomplete migration plan error for virtual machines with shared disks before cold migration
```
@RichardHoch @anarnold97 could you review this new resolve issue, please?

Pkylas007 commented:
@fabiand @mnecas Could you provide your comments for this PR, please?

fabiand commented on documentation/modules/known-issues-2-8.adoc:
```diff
@@ -0,0 +1,17 @@
+
+{project-first} 2.8 has the following known issues:
+
+.Target VMware virtual machines lack connectivity because of unassigned static IP address
```
This issue got fixed. No need for a known issue.

fabiand commented on documentation/modules/known-issues-2-8.adoc:
```diff
@@ -0,0 +1,17 @@
+.Target VMware virtual machines lack connectivity because of unassigned static IP address
+After migrating Ubuntu 20.04 VMware virtual machines (VM) to {virt} 4.15.6 using {project-short}, the  network interface card (NIC) name of the source VM is not preserved. This results in a lack of network connectivity for the VM because the network interface gets a new name post migration. The static IP address that was configured in the network plan before migration is not assigned to the new n ...
+
+.Persistent volume claims are populated by incorrect disks
```
This also got fixed in 2.8. It can be removed

fabiand commented on documentation/modules/new-features-and-enhancements-2-8.adoc:
```diff
@@ -0,0 +1,17 @@
+[id="new-features-and-enhancements-2-8-0_{context}"]
+== New features and enhancements 2.8.0
+
+* In {project-short} 2.8.0, warm migration is now based on RHEL 9 inheriting features and bug fixes.
```
this can be removed, this was part of 2.7.

Pkylas007 commented:
@fabiand thank you for your comments! I have moved the two known issues to resolved issues if that's fine. Also, removed the 2.7 content. Please can you check and confirm is the updates are ok?

yaacov commented on documentation/modules/known-issues-2-8.adoc:
```diff
@@ -0,0 +1,14 @@
+After warm migration of Windows Server 2022 virtual machine (VM) from vSphere to an {ocp-name} cluster, the Trusted Platform Module (TPM) device is added in the target VM, even if the TPM configuration is non-persistent in the BIOS setting of the source Windows Server 2022 VM. This prevents the target VM from starting. link:https://issues.redhat.com/browse/MTV-2049[(MTV-2049)]
+
+.Linux virtual machines with clustered file systems are not supported
+Linux guest virtual machines with clustered file systems on a shared disk are not supported. The virt-v2v tool may not update shared disks in the `fstab`configuration file. link:https://issues.redhat.com/browse/MTV-2145[(MTV-2145)]
```
```suggestion
Linux guest virtual machines with clustered file systems on a shared disk are not supported. The virt-v2v tool may not update shared disks in the `fstab` configuration file. link:https://issues.redhat.com/browse/MTV-2145[(MTV-2145)]
```

Pkylas007 replied:
Thank you @yaacov , made the change in the document.

yaacov commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+In earlier releases of {project-short}, for migration plans in which `Preserve static IPs` is enabled, {project-short} displayed an error message if you did not configure a static IP address for the guest virtual machine. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-1458[(MTV-1458)]
+
+.Disk counter does not update progress while migrating virtual machine with multiple disks
+In earlier releases of {project-short}, the `Disk Counter`did not show real-time progress of disk transfer when migrating a virtual machine with multiple disks. This issue has been resolved in {project-short} 2.8.0.link:https://issues.redhat.com/browse/MTV-1718[(MTV-1718)]
```
```suggestion
In earlier releases of {project-short}, the `Disk Counter` did not show real-time progress of disk transfer when migrating a virtual machine with multiple disks. This issue has been resolved in {project-short} 2.8.0.link:https://issues.redhat.com/browse/MTV-1718[(MTV-1718)]
```

anarnold97 commented on documentation/modules/known-issues-2-8.adoc:
```diff
@@ -0,0 +1,14 @@
+.Trusted Platform Module is added to BIOS-based Windows Server 2022 VMs after warm migration
+After warm migration of Windows Server 2022 virtual machine (VM) from vSphere to an {ocp-name} cluster, the Trusted Platform Module (TPM) device is added in the target VM, even if the TPM configuration is non-persistent in the BIOS setting of the source Windows Server 2022 VM. This prevents the target VM from starting. link:https://issues.redhat.com/browse/MTV-2049[(MTV-2049)]
+
+.Linux virtual machines with clustered file systems are not supported
```
```suggestion
.Linux VMs with clustered file systems are not supported
```  
as you used `VMs` in the previous header, i would use the same pattern here

anarnold97 commented on documentation/modules/known-issues-2-8.adoc:
```diff
@@ -0,0 +1,14 @@
+After warm migration of Windows Server 2022 virtual machine (VM) from vSphere to an {ocp-name} cluster, the Trusted Platform Module (TPM) device is added in the target VM, even if the TPM configuration is non-persistent in the BIOS setting of the source Windows Server 2022 VM. This prevents the target VM from starting. link:https://issues.redhat.com/browse/MTV-2049[(MTV-2049)]
+
+.Linux virtual machines with clustered file systems are not supported
+Linux guest virtual machines with clustered file systems on a shared disk are not supported. The virt-v2v tool may not update shared disks in the `fstab` configuration file. link:https://issues.redhat.com/browse/MTV-2145[(MTV-2145)]
```
```suggestion
Linux guest VMs with clustered file systems on a shared disk are not supported. The virt-v2v tool may not update shared disks in the `fstab` configuration file. link:https://issues.redhat.com/browse/MTV-2145[(MTV-2145)]
```

as you have explained the acronym in line 11: `...virtual machine (VM)... `
i think you are ok to use the acronym here

anarnold97 commented on documentation/modules/known-issues-2-8.adoc:
```diff
@@ -0,0 +1,14 @@
+{project-first} 2.8 has the following known issues:
+
+.Trusted Platform Module is added to BIOS-based Windows Server 2022 VMs after warm migration
+After warm migration of Windows Server 2022 virtual machine (VM) from vSphere to an {ocp-name} cluster, the Trusted Platform Module (TPM) device is added in the target VM, even if the TPM configuration is non-persistent in the BIOS setting of the source Windows Server 2022 VM. This prevents the target VM from starting. link:https://issues.redhat.com/browse/MTV-2049[(MTV-2049)]
```
```suggestion
After warm migration of Windows Server 2022 virtual machine (VM) from vSphere to an {ocp-name} cluster, the Trusted Platform Module (TPM) device is added to the target VMs, even if the TPM configuration is non-persistent in the BIOS setting of the source Windows Server 2022 VM. This prevents the target VM from starting. link:https://issues.redhat.com/browse/MTV-2049[(MTV-2049)]
```

anarnold97 commented on documentation/modules/known-issues-2-8.adoc:
```diff
@@ -0,0 +1,14 @@
+
+{project-first} 2.8 has the following known issues:
+
+.Trusted Platform Module is added to BIOS-based Windows Server 2022 VMs after warm migration
```
```suggestion
.Trusted Platform Module is added to the BIOS of a Windows Server 2022 VM after warm migration
```
not too sure if we should use `BIOS-based`

anarnold97 commented on documentation/modules/new-features-and-enhancements-2-8.adoc:
```diff
@@ -0,0 +1,15 @@
+[id="new-features-and-enhancements-2-8-0_{context}"]
+== New features and enhancements 2.8.0
+
+* In {project-short} 2.8.0, you can provide custom names for the following attributes in the migration plan of VMware virtual machines (VM):
```
```suggestion
* In {project-short} 2.8.0, you can provide custom names for the following attributes in the migration plan of VMware virtual machines (VMs):
```
mis-match in acronym

anarnold97 commented on documentation/modules/new-features-and-enhancements-2-8.adoc:
```diff
@@ -0,0 +1,15 @@
+* In {project-short} 2.8.0, you can provide custom names for the following attributes in the migration plan of VMware virtual machines (VM):
++ 
+** Persistent volume claim (PVC) name template used to generate PVC names in target VM. 
+** Volume name template used to generate volume interface names for target VMs.
```
```suggestion
** Volume name template is used to generate volume interface names for the target VMs.
```

anarnold97 commented on documentation/modules/new-features-and-enhancements-2-8.adoc:
```diff
@@ -0,0 +1,15 @@
++ 
+** Persistent volume claim (PVC) name template used to generate PVC names in target VM. 
+** Volume name template used to generate volume interface names for target VMs.
+** Network name template used to generate network interface names for target VMs.
```
```suggestion
** Network name template is used to generate network interface names for the target VMs.
```
(pedantic)
write in patterns:
* some of these lines are `... is used` and others are just `... used` 
* i would add the article... but meh

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+[id="resolved-issues-2-8-0_{context}"]
+== Resolved issues 2.8.0
+
+.Target VMware virtual machines lack connectivity because of unassigned static IP address
```
```suggestion
.Target VMware virtual machines lack connectivity because of an unassigned static IP address
```

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+== Resolved issues 2.8.0
+
+.Target VMware virtual machines lack connectivity because of unassigned static IP address
+In earlier releases of {project-short}, after migrating Ubuntu 20.04 VMware virtual machines (VM) to {virt} 4.15.6 using {project-short}, the  network interface card (NIC) name of the source VM was not preserved. This resulted in a lack of network connectivity for the VM because the network interface was assigned a new name post migration. The static IP address configured in the network plan befo ...
```
```suggestion
In earlier releases of {project-short}, after migrating Ubuntu 20.04 VMware virtual machines (VM) to {virt} 4.15.6 using {project-short}, the  network interface card (NIC) name of the source VM was not preserved. This resulted in a lack of network connectivity for the VM because the network interface was assigned a new name post-migration. The static IP address configured in the network plan before migration was not assigned to the new network interface. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-1607[(MTV-1607)]
```
i would probably use a compound

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+In earlier releases of {project-short}, after migrating Ubuntu 20.04 VMware virtual machines (VM) to {virt} 4.15.6 using {project-short}, the  network interface card (NIC) name of the source VM was not preserved. This resulted in a lack of network connectivity for the VM because the network interface was assigned a new name post migration. The static IP address configured in the network plan befo ...
+
+.MTV does not preserve out of range gateway IP address 
+In earlier releases of {project-short}, the gateway IP address of the virtual machine was not copied after migration if the IP address was out of network interface subnet range. The issue affected migrations even if you enabled the `Preserve static IPs` setting. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-2018[(MTV-2018)]
```
```suggestion
In earlier releases of {project-short}, the gateway IP address of the virtual machine was not copied after migration if the IP address was out of network interface subnet range. This issue affected migrations even if you enabled the `Preserve static IPs` setting. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-2018[(MTV-2018)]
```

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+.Target VMware virtual machines lack connectivity because of unassigned static IP address
+In earlier releases of {project-short}, after migrating Ubuntu 20.04 VMware virtual machines (VM) to {virt} 4.15.6 using {project-short}, the  network interface card (NIC) name of the source VM was not preserved. This resulted in a lack of network connectivity for the VM because the network interface was assigned a new name post migration. The static IP address configured in the network plan befo ...
+
+.MTV does not preserve out of range gateway IP address 
```
```suggestion
.MTV does not preserve the gateway of a NIC from different subnet
```
i might just simplify this using Martin's [PR](https://github.com/kubev2v/forklift/pull/1378)

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+.MTV does not preserve out of range gateway IP address 
+In earlier releases of {project-short}, the gateway IP address of the virtual machine was not copied after migration if the IP address was out of network interface subnet range. The issue affected migrations even if you enabled the `Preserve static IPs` setting. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-2018[(MTV-2018)]
+
+.MTV displays an error in the migration plan for virtual machines without a static IP address
```
```suggestion
.MTV displays an error in the migration plan for VMs without a static IP address
```

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+In earlier releases of {project-short}, the gateway IP address of the virtual machine was not copied after migration if the IP address was out of network interface subnet range. The issue affected migrations even if you enabled the `Preserve static IPs` setting. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-2018[(MTV-2018)]
+
+.MTV displays an error in the migration plan for virtual machines without a static IP address
+In earlier releases of {project-short}, for migration plans in which `Preserve static IPs` is enabled, {project-short} displayed an error message if you did not configure a static IP address for the guest virtual machine. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-1458[(MTV-1458)]
```
```suggestion
In earlier releases of {project-short}, for migration plans in which `Preserve static IPs` is enabled, {project-short} displayed an error message if you did not configure a static IP address for the guest VM. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-1458[(MTV-1458)]
```

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+.MTV displays an error in the migration plan for virtual machines without a static IP address
+In earlier releases of {project-short}, for migration plans in which `Preserve static IPs` is enabled, {project-short} displayed an error message if you did not configure a static IP address for the guest virtual machine. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-1458[(MTV-1458)]
+
+.Disk counter does not update progress while migrating virtual machine with multiple disks
```
```suggestion
.Disk counter does not update progress while migrating VMs with multiple disks
```

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+In earlier releases of {project-short}, for migration plans in which `Preserve static IPs` is enabled, {project-short} displayed an error message if you did not configure a static IP address for the guest virtual machine. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-1458[(MTV-1458)]
+
+.Disk counter does not update progress while migrating virtual machine with multiple disks
+In earlier releases of {project-short}, the `Disk Counter` did not show real-time progress of disk transfer when migrating a virtual machine with multiple disks. This issue has been resolved in {project-short} 2.8.0.link:https://issues.redhat.com/browse/MTV-1718[(MTV-1718)]
```
```suggestion
In earlier releases of {project-short}, the `Disk Counter` did not show real-time progress of disk transfer when migrating a VM with multiple disks. This issue has been resolved in {project-short} 2.8.0.link:https://issues.redhat.com/browse/MTV-1718[(MTV-1718)]
```

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+.Disk counter does not update progress while migrating virtual machine with multiple disks
+In earlier releases of {project-short}, the `Disk Counter` did not show real-time progress of disk transfer when migrating a virtual machine with multiple disks. This issue has been resolved in {project-short} 2.8.0.link:https://issues.redhat.com/browse/MTV-1718[(MTV-1718)]
+
+.Incomplete migration plan error for virtual machines with shared disks before cold migration
```
```suggestion
.Incomplete migration plan error for VMs with shared disks before cold migration
```

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+In earlier releases of {project-short}, the `Disk Counter` did not show real-time progress of disk transfer when migrating a virtual machine with multiple disks. This issue has been resolved in {project-short} 2.8.0.link:https://issues.redhat.com/browse/MTV-1718[(MTV-1718)]
+
+.Incomplete migration plan error for virtual machines with shared disks before cold migration
+In earlier releases of {project-short}, the migration plan for a virtual machine (VM) with shared disks showed an error that did not list all VMs to be powered off before the cold migration. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-2206[(MTV-2206)]
```
```suggestion
In earlier releases of {project-short}, the migration plan for a VM with shared disks showed an error that did not list all VMs to be powered off before the cold migration. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-2206[(MTV-2206)]
```

Pkylas007 commented on documentation/modules/new-features-and-enhancements-2-8.adoc:
```diff
@@ -0,0 +1,15 @@
+
+* In {project-short} 2.8.0, you can provide custom names for the following attributes in the migration plan of VMware virtual machines (VM):
++ 
+** Persistent volume claim (PVC) name template used to generate PVC names in target VM. 
```
Thank you Andy! Should bullet points be complete sentences? I think we can discuss this.

mnecas commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+In earlier releases of {project-short}, for migration plans in which `Preserve static IPs` is enabled, {project-short} displayed an error message if you did not configure a static IP address for the guest VM. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-1458[(MTV-1458)]
+
+.Disk counter does not update progress while migrating VMs with multiple disks
+In earlier releases of {project-short}, the `Disk Counter` did not show real-time progress of disk transfer when migrating a VM with multiple disks. This issue has been resolved in {project-short} 2.8.0.link:https://issues.redhat.com/browse/MTV-1718[(MTV-1718)]
```
NP: Please add space between the `2.8.0.` and link

mnecas reviewed (changes requested):
I went through the 2.8.0 list and could you please also add:
- New feature https://issues.redhat.com/browse/MTV-1744
- Bug fix https://issues.redhat.com/browse/MTV-1796 for this customer is waiting so think it's better to mention it in the release notes.
You are 100% right about adding Release-Docs label.

Pkylas007 commented:
> I went through the 2.8.0 list and could you please also add:
> 
> * New feature https://issues.redhat.com/browse/MTV-1744
> * Bug fix https://issues.redhat.com/browse/MTV-1796 for this customer is waiting so think it's better to mention it in the release notes.
>   You are 100% right about adding Release-Docs label.

Hi @mnecas, thank you for sharing the tickets! I have updated the doc with the new feature and two resolved issues (the final two items in Resolved Issues). Could you check again and let me know if the updates look alright?

mnecas commented on documentation/modules/mtv-selected-packages-2-8.adoc:
```diff
@@ -0,0 +1,17 @@
+.Selected {project-short} packages
+[width="100%",cols="50%,50%,options="header",]
+|===
+|Package summary
```
This section will be added later?

Pkylas007 replied:
Yes @mnecas , this content will be added later.

nunzy1 commented on documentation/modules/known-issues-2-8.adoc:
```diff
@@ -0,0 +1,14 @@
+After warm migration of Windows Server 2022 virtual machine (VM) from vSphere to an {ocp-name} cluster, the Trusted Platform Module (TPM) device is added to the target VMs, even if the TPM configuration is non-persistent in the BIOS setting of the source Windows Server 2022 VM. This prevents the target VM from starting. link:https://issues.redhat.com/browse/MTV-2049[(MTV-2049)]
+
+.Linux VM with clustered file systems are not supported
+Linux guest VMs with clustered file systems on a shared disk are not supported. The `virt-v2v` tool may not update shared disks in the `fstab` configuration file. link:https://issues.redhat.com/browse/MTV-2145[(MTV-2145)]
```
Can we say "does not update" rather than "may"?

nunzy1 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+== Resolved issues 2.8.0
+
+.MTV does not preserve the gateway IP address of an NIC from a different subnet
+In earlier releases of {project-short}, the gateway IP address of the virtual machine was not copied after migration if the IP address was out of network interface subnet range. This issue affected migrations even if you enabled the `Preserve static IPs` setting. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-2018[(MTV-2018)]
```
"was out of network interface subnet range"  Add "the".

Pkylas007 replied:
Thank you!

nunzy1 commented on documentation/modules/technical-changes-2-8.adoc:
```diff
@@ -0,0 +1,19 @@
+{project-short} detects shared disks used by multiple VMware virtual machines (VMs) in the migration plan and provides an option to migrate the shared disks one time. Migration of shared disks is supported for VMs created using the following operating systems:
+
+* Windows
+* Linux block devices only
```
Linux (block devices only)

mnecas reviewed (approved):
lgtm

anarnold97 commented on documentation/modules/new-features-and-enhancements-2-8.adoc:
```diff
@@ -0,0 +1,17 @@
+** Volume name template that is used to generate volume interface names for the target VMs.
+** Network name template that is used to generate network interface names for the target VMs.
+
+* In {project-short} 2.8.0, VM names can contain periods as described in link:https://datatracker.ietf.org/doc/html/rfc1123[(RFC-1123)]. This allows you to use Fully Qualified Domain Names in VM names after migration. 
```
```suggestion
* In {project-short} 2.8.0, VM names can contain periods, as described in link:https://datatracker.ietf.org/doc/html/rfc1123[(RFC-1123)]. This allows you to use Fully Qualified Domain Names in VM names after migration. 
```
sorry - pedantic

Pkylas007 replied:
:) thanks! added the comma here

anarnold97 commented on documentation/modules/rn-28-resolved-issues.adoc:
```diff
@@ -0,0 +1,29 @@
+In earlier releases of {project-short}, during warm migrations of multiple VMware VMs, {project-short} continued to run the migration plan after you canceled a VM in the plan. The initial plan was run after you successfully migrated the canceled VM through a second migration plan. This issue has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-1796[(MTV-1796)]
+
+.Static IP address is not preserved for VMs with manual MAC address configuration
+In earlier releases of {project-short}, the static IP address of a VMware VM was not preserved if you configured a MAC address with upper case for the virtual network adapter of the VM. This issue affected migrations where the `Preserve static IPs` setting was enabled and has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-2285[(MTV-2285)] 
```
```suggestion
In earlier releases of {project-short}, the static IP address of a VMware VM was not preserved if you configured a MAC address with uppercase for the virtual network adapter of the VM. This issue affected migrations where the `Preserve static IPs` setting was enabled and has been resolved in {project-short} 2.8.0. link:https://issues.redhat.com/browse/MTV-2285[(MTV-2285)] 
```

Pkylas007 replied:
Thank you! Corrected it.

anarnold97 reviewed (changes requested):
Looks v.good... but saw two pedantic changes (for which i apologize)

Pkylas007 commented:
> Looks v.good... but saw two pedantic changes (for which i apologize)

Thank you for going through the draft multiple times! I made all the suggested changes.

Pkylas007 commented:
> lgtm

Thank you @mnecas !
